lib/Net/SIP/Dispatcher.pm

```diff
@@ -861,24 +861,36 @@ sub __resolve_uri_final {
     # FIXME: can contradict order in @proto
     @$resp = sort { $a->{prio} <=> $b->{prio} } @$resp;
 
+    # remove duplicate entries in response
+    my %resp;
+    for(my $i=0; $i<@$resp; $i++) {
+	my $r = $resp->[$i];
+	if (exists $resp{$r->{host}}->{$r->{family}}->{$r->{proto}}->{$r->{addr}}->{$r->{port}}) {
+	    splice(@$resp,$i,1);
+	} else {
+	    $resp{$r->{host}}->{$r->{family}}->{$r->{proto}}->{$r->{addr}}->{$r->{port}} = $r->{prio};
+	}
+    }
+
     @$dst_addr = ();
-    @$legs = ();
+    my %legs;
     foreach my $r ( @$resp ) {
-	my $leg = first { $_->can_deliver_to(
+	my @legs = grep { $_->can_deliver_to(
 	    proto  => $r->{proto},
 	    host   => $r->{host},
 	    addr   => $r->{addr},
 	    port   => $r->{port},
 	    family => $r->{family},
 	)} @$allowed_legs;
 
-	if ( $leg ) {
+	if ( @legs ) {
 	    push @$dst_addr, $r;
-	    push @$legs,$leg;
+	    $legs{refaddr($_)} = $_ foreach @legs;
 	} else {
 	    DEBUG(50,"no leg with $r->{proto} to %s", ip_parts2string($r));
 	}
     }
+    @$legs = values %legs;
```

Comment:
The order returned by `values %legs` is unpredictable and can change with each run of this part of code. As far as I can see the previous code was more predictable in what it did.

Reply:
Fixed. Now I changed this code to be fully deterministic as before.